junit/src/main/java/net/neoforged/gradle/junit/JUnitProjectPlugin.java

```diff
@@ -0,0 +1,157 @@
+        project.afterEvaluate(this::applyAfterEvaluate);
+    }
+
+    private void applyAfterEvaluate(Project project) {
+        Test testTask = (Test) project.getTasks().getByName("test");
+        if (!(testTask.getTestFrameworkProperty().get() instanceof JUnitPlatformTestFramework)) {
+            LOG.info("Ignoring test task {} because it doesn't use JUnit 5", testTask.getName());
+            return;
+        }
+
+        configureTestTask(testTask, "junit");
+    }
+
+    /**
+     * Configures a JUnit Test-Task to have the required system properties, environment variables, jvm arguments
+     * and task dependencies to run the JUnit support code from fml-junit.
+     *
+     * @param runType The name of the {@link net.neoforged.gradle.dsl.common.runs.type.RunType} to read
+     *                the properties from.
+     */
+    private static void configureTestTask(Test testTask, String runType) {
+        TestTaskExtension extension = testTask.getExtensions().getByType(TestTaskExtension.class);
+        Project project = testTask.getProject();
+
+        // Find the runtime reachable from the test task
+        @Nullable CommonRuntimeDefinition<?> runtimeDefinition;
+        try {
+            runtimeDefinition = TaskDependencyUtils.extractRuntimeDefinition(project, testTask);
+        } catch (MultipleDefinitionsFoundException e) {
+            throw new GradleException("Couldn't determine Minecraft runtime definition for JUnit tests", e);
+        } catch (NoDefinitionsFoundException e) {
+            // Try again to see if it's a forge-dev project
+            Object runtimeExtension = project.getExtensions().findByName("runtime");
+            if (runtimeExtension instanceof CommonRuntimeDefinition) {
+                runtimeDefinition = (CommonRuntimeDefinition<?>) runtimeExtension;
+            } else {
+                runtimeDefinition = null;
+            }
+        }
+
+        // If it is a userdev runtime, we automatically add the additional testing libraries declared by the runtime
+        // this will add the FML JUnit support library without the user having to declare it themselves.
+        Configuration testRuntimeOnly = project.getConfigurations().getByName("testRuntimeOnly");
+        DependencyFactory dependencyFactory = project.getDependencyFactory();
+        if (runtimeDefinition instanceof UserDevRuntimeDefinition) {
+            UserDevRuntimeDefinition userdevRuntime = (UserDevRuntimeDefinition) runtimeDefinition;
+            List<String> testDependencies = userdevRuntime.getUserdevConfiguration().getAdditionalTestDependencyArtifactCoordinates().get();
+            for (String testDependency : testDependencies) {
+                testRuntimeOnly.getDependencies().add(dependencyFactory.create(testDependency));
+            }
+        }
+
+        RunImpl junitRun = getOrCreateRun("junitTest", project, runtimeDefinition, runType);
```

Comment:
No if the user does not create one, then you should abort processing.
The userdev runtime will create a run named after the runtype, so creating another run here is redundant and wrong anyway.

Reply:
No, it does not create them automatically. And even if it did, that would be a bad idea for the junit run type (and so would the user creating it normally). The run would generate `run*` tasks, `prepare*` tasks, and IDE run configurations which wouldn't work with JUnit.

Reply:
Indeed it does not create the runs automatically. But your runtype should be configured enough that it should only need a: 
```groovy
runs {
   unitTests{} //Or what ever the name of the unit test run type is.
}
```
To get the plugin to work. That is how all other run types are designed.
Please redesign this area.